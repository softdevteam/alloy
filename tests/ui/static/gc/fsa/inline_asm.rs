#![feature(gc)]
#![feature(negative_impls)]
#![feature(rustc_private)]
#![allow(dead_code)]
#![allow(unused_variables)]
include!{"./auxiliary/types.rs"}

use std::arch::asm;

#[derive(Debug)]
struct ASM;

impl Drop for ASM {
    fn drop(&mut self) {
        let a: u64 = 10;
        let b: u64 = 20;
        let result: u64;
        unsafe {
            asm!(
                "add {0}, {1}, {2}",
                out(reg) result,
                in(reg) a,
                in(reg) b
            );
        }
    }
}


fn main() {
    Gc::new(FinalizerUnsafeWrapper(ASM));
    //~^ ERROR: The drop method for `ASM` cannot be safely finalized.
}
