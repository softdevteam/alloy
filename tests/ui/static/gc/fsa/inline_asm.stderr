error: The drop method for `ASM` cannot be safely finalized.
  --> $DIR/inline_asm.rs:31:13
   |
LL | /             asm!(
LL | |                 "add {0}, {1}, {2}",
LL | |                 out(reg) result,
LL | |                 in(reg) a,
LL | |                 in(reg) b
LL | |             );
   | |_____________- this assembly block is not safe to run in a finalizer
...
LL |       Gc::new(FinalizerUnsafeWrapper(ASM));
   |       --------^^^^^^^^^^^^^^^^^^^^^^^^^^^- caused by trying to construct a `Gc<FinalizerUnsafeWrapper<ASM>>` here.

error: aborting due to 1 previous error

